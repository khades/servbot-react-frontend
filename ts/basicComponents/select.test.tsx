import * as React from "react";
import { create } from "react-test-renderer";
import Select, { ISelectProps } from "./select";

describe("Select", () => {
    it("should render and have no errors and no options selected", () => {
        const props: ISelectProps = {
            getValue: () => "",
            getValues: () => [{ value: "1", label: "one" }, { value: "2", label: "two" }],
            id: "ID",
            label: "LABEL",
            setValue: (input: string) => input,
        };

        const select = create(
            <Select {...props} />,
        );
        expect(select.root.findAll((el) => el.type === "option").length).toBe(3);
        expect(select.root.findAll(
            (el) => el.type === "option" && el.props.disabled === true).length).toBe(1);
        expect(select.root.findAll(
            (el) => el.type === "option").length).toBe(3);
        expect(select.toJSON()).toMatchSnapshot();

    });

    it("should render and have no errors and option selected", () => {
        const props: ISelectProps = {
            getValue: () => "1",
            getValues: () => [{ value: "1", label: "one" }, { value: "2", label: "two" }],
            id: "ID",
            label: "LABEL",
            setValue: (input: string) => input,
        };

        const select = create(
            <Select {...props} />,
        );
        expect(select.root.findAll((el) => el.type === "option").length).toBe(2);
        expect(select.toJSON()).toMatchSnapshot();

    });

    it("value change should work", () => {
        let selectValue = "";
        const props: ISelectProps = {
            getValue: () => selectValue,
            getValues: () => [{ value: "1", label: "one" }, { value: "2", label: "two" }],
            id: "ID",
            label: "LABEL",
            setValue: (value: string) => selectValue = value,
        };

        const select = create(
            <Select {...props} />,
        );
        expect(select.root.findAll(
            (el) => el.type === "option" && el.props.disabled === true).length).toBe(1);

        expect(select.root.findAll((el) => el.type === "option").length).toBe(3);
        const selectInstance = select.root.find((el) => el.type === "select");
        selectInstance.props.onChange({ currentTarget: { value: "2" } });
        select.update(
            <Select {...props} />,
        );
        expect(select.root.findAll((el) => el.type === "option").length).toBe(2);
        expect(selectValue).toBe("2");
    });
});
