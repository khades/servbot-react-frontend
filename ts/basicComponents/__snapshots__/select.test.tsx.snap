// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render and have no errors and no options selected 1`] = `
<div
  className="control-group"
>
  <label
    className="control-group__label"
    htmlFor="ID"
  >
    LABEL
  </label>
  <select
    defaultValue=""
    id="ID"
    onChange={[Function]}
  >
    <option
      disabled={true}
      value=""
    />
    <option
      value="1"
    >
      one
    </option>
    <option
      value="2"
    >
      two
    </option>
  </select>
</div>
`;

exports[`Select should render and have no errors and option selected 1`] = `
<div
  className="control-group"
>
  <label
    className="control-group__label"
    htmlFor="ID"
  >
    LABEL
  </label>
  <select
    defaultValue="1"
    id="ID"
    onChange={[Function]}
  >
    <option
      value="1"
    >
      one
    </option>
    <option
      value="2"
    >
      two
    </option>
  </select>
</div>
`;
